pkgs/servers/peertube/default.nix

```diff
@@ -13,14 +13,28 @@
 , yarn
 }:
 let
-  arch =
-    if stdenv.hostPlatform.system == "x86_64-linux" then "linux-x64"
+  bcryptAttrs =
+    if stdenv.hostPlatform.system == "x86_64-linux" then {
+      arch = "linux-x64";
+      libc = "glibc";
+      hash = "sha256-3R3dBZyPansTuM77Nmm3f7BbTDkDdiT2HQIrti2Ottc=";
+    }
+    else if stdenv.hostPlatform.system == "x86_64-darwin" then {
+      arch = "darwin-x64";
+      libc = "unknown";
+      hash = "sha256-r8cD52t1quWV0gBDLyEaa3cO4zOKrwR299Vfntnpyzw=";
+    }
+    else if stdenv.hostPlatform.system == "aarch64-darwin" then {
+      arch = "darwin-arm64";
+      libc = "unknown";
+      hash = "sha256-VrZKQIc1oG+bgBKV11WoNq3Ny09tngKPLTJ6swW5FUg=";
+    }
     else throw "Unsupported architecture: ${stdenv.hostPlatform.system}";
```

Comment:
This should be solved via an attrset rather than chained if else.

Reply:
Done, thank you for the suggestion.